Assets/Scripts/UI/UI_TileInfoManager.cs

```diff
@@ -19,10 +19,20 @@ private void RefreshTileInfoUI()
             {
                 _previousTerrainData = ActiveTerrainData;
 
-                _terrainValueText.text = ActiveTerrainData.TerrainType.ToString();
-                _defenseValueText.text = ActiveTerrainData.DefenseBoost.ToString();
-                _movementValueText.text = ActiveTerrainData.MovementCost.ToString();
-                _healthBuffValueText.text = ActiveTerrainData.HealPercentageBoost.ToString();
+                if (ActiveTerrainData.TerrainType == Constants.Terrain.Unwalkable)
+                {
+                    _terrainValueText.text = ActiveTerrainData.TerrainType.ToString();
```

Comment:
terrain type value is same in both if/else. Can be outside checks to avoid duplication.